src/Extensions/Features/src/FeatureCollectionExtensions.cs

```diff
@@ -0,0 +1,52 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.AspNetCore.Http.Features;
+
+/// <summary>
+/// Extension methods for getting feature from <see cref="IFeatureCollection"/>
+/// </summary>
+public static class FeatureCollectionExtensions
+{
+    /// <summary>
+    /// Get feature of type <typeparamref name="TFeature"/> from the <see cref="IFeatureCollection"/>
+    /// </summary>
+    /// <param name="featureCollection"></param>
+    /// <returns>Feature object type</returns>
+    /// <exception cref="InvalidOperationException">Thrown when asked for unregistered feature type</exception>
```

Comment:
We don't document exceptions elsewhere. I would err on the side of consistency and skip this.

Reply:
At least mention the exception in the summary.